mlflow/utils/mlflow_tags.py

```diff
@@ -80,7 +82,7 @@
 LATEST_CHECKPOINT_ARTIFACT_TAG_KEY = "mlflow.latest_checkpoint_artifact"
 
 # A set of tags that cannot be updated by the user
-IMMUTABLE_TAGS = {MLFLOW_USER, MLFLOW_ARTIFACT_LOCATION}
+IMMUTABLE_TAGS = {MLFLOW_USER, MLFLOW_ARTIFACT_LOCATION, TRACE_SCHEMA_VERSION_KEY}
```

Comment:
oh wait i just realized this is not a tag, but in `request_metadata`, so this won't work. cc @serena-ruan do you know what's the difference between how we use `request_metadata` and `tags`? maybe trace schema version can be included in the trace info's tags instead.